src/main/java/net/neoforged/neoforge/client/ClientHooks.java

```diff
@@ -1033,6 +1037,17 @@ public static void onCreativeModeTabBuildContents(CreativeModeTab tab, ResourceK
             output.accept(entry.getKey(), entry.getValue());
     }
 
+    private static final BiMap<ResourceLocation, SpriteSourceType> SPRITE_SOURCE_TYPES_MAP = HashBiMap.create();
+
+    public static BiMap<ResourceLocation, SpriteSourceType> makeSpriteSourceTypesMap() {
+        return SPRITE_SOURCE_TYPES_MAP;
+    }
+
+    @ApiStatus.Internal
+    public static void registerSpriteSourceTypes() {
+        ModLoader.get().postEvent(new RegisterSpriteSourceTypesEvent(SPRITE_SOURCE_TYPES_MAP));
```

Comment:
This event only fires exactly once on startup. Clearing the map would also nuke the vanilla entries since they are statically registered to the same map

Reply:
can't do that the map is the same vanilla register to in static init